model/sort_traces.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package model
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+)
+
+type spanBySpanID []*Span
+
+func (s spanBySpanID) Len() int           { return len(s) }
+func (s spanBySpanID) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s spanBySpanID) Less(i, j int) bool { return s[i].SpanID < s[j].SpanID }
+
+// SortTraces checks two traces' length and sorts their spans.
+func SortTraces(expected *Trace, actual *Trace) error {
```

Comment:
Hmm, so I did this before, and the issue with this was that I would get index out of bound error with the assert.Equals, and the error message wouldn't be very helpful. the current way catches these because they'll do a length check on the way